modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,54 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
+  @derive(eqv, customizableDecoder, customizableEncoder, show)
   case class Amount(
     value: AmountValue,
     currency: Currency
-  )
+  ) {
+    def negate: Either[String, Amount] =
+      value.negate.map(Amount(_, currency))
+  }
 
-  type AmountValueRefined = Long Refined Not[Equal[0L]]
+  type AmountValuePredicate = Not[Equal[0L]]
+  type AmountValueRefined = Long Refined AmountValuePredicate
 
-  @derive(eqv, decoder, encoder)
+  @derive(eqv, decoder, encoder, show)
   @newtype
   case class AmountValue(value: AmountValueRefined) {
     def isNegative: Boolean = value < 0L
+    def isPositive: Boolean = value > 0L
+
+    def negate: Either[String, AmountValue] =
+      refineV[AmountValuePredicate](-value.value).map(rv => AmountValue(rv))
```

Comment:
is there a correct `AmountValue` that cannot be negated?

Reply:
@marcinwadon

1. `negate(- Long.MinValue) == - Long.MinValue`  Is this worth keeping it returning an _Either_?
2. If we ignore 1. then how can I produce a runtime _Refined_ that doesn't produce an _Either_? Use `Refined.unsafeApply`?

Reply:
I guess if you are sure that every `AmountValueRefined` can be negated then you can use unsafe for that case to avoid getting a surplus or obfuscating (in that scenario) Either back.

Reply:
I guess there is a problem with `minLong` which can't be negated right? Negated `minLong` should give a value of `maxLong` + 1 which is impossible.

Reply:
@TheMMaciek Yes I believe so. It doesn't "fail" but the result is not what's expected.